src/creature.cpp

```diff
@@ -614,6 +614,23 @@ void Creature::onCreatureMove(Creature* creature, const Tile* newTile, const Pos
 	}
 }
 
+std::vector<Creature*> Creature::getKillers()
+{
+	std::vector<Creature*> killers;
+	killers.reserve(damageMap.size());
+	const int64_t timeNow = OTSYS_TIME();
+	const uint32_t inFightTicks = g_config.getNumber(ConfigManager::PZ_LOCKED);
+	for (const auto& it : damageMap) {
+		if (Creature* attacker = g_game.getCreatureByID(it.first)) {
+			CountBlock_t cb = it.second;
+			if (attacker != this && timeNow - cb.ticks <= inFightTicks) {
+				killers.push_back(attacker);
+			}
+		}
+	}
```

Comment:
Hint: this is [`std::copy_if`](https://en.cppreference.com/w/cpp/algorithm/copy)

Reply:
I was trying your `copy_if` suggestion but it only works for two vectors of the same type!
so I made the decision to use `for_each`